bindings/python/scripts/generate.py

```diff
@@ -1,105 +1,284 @@
-        else:
-            module_linelist[i] = "".join(
-                (f"PYBIND11_MODULE({module_name}, m)", "{", module_linelist[i])
-            )
-            break
-    for line in module_linelist:
-        linelist.append(line)
-    linelist.append("}")
-    return linelist
+        self._state_stack.append(
+            {"kind": self.kind, "name": self.name, "depth": self.depth}
+        )
 
+        self.kind_functions[self.kind]()
 
-def handle_include(item):
-    pass
+        if self.kind_functions[self.kind] is not self.skip:
+            for sub_item in self.members:
+                self.handle_node(sub_item)
 
+        self.close()
 
-def handle_alias(item):
-    pass
+        self._state_stack.pop()
 
+    def handle_namespace(self):
+        self.linelist.append(f"namespace {self.name}" + "{")
 
-def handle_constructor(item):
-    parameters_type = ""
-    parameters_type = ",".join(params for params in item["parameters_type"])
-    module_linelist.append(f".def(py::init<{parameters_type}>())")
+    def get_fields_from_anonymous(self, item):
+        fields = []
+        for sub_item in item["members"]:
+            if sub_item["kind"] in ("ANONYMOUS_UNION_DECL", "ANONYMOUS_STRUCT_DECL"):
+                for field in self.get_fields_from_anonymous(sub_item):
+                    fields.append(field)
+            if sub_item["kind"] == "FIELD_DECL":
+                fields.append(sub_item)
+        return fields
 
+    def handle_struct_decl(self):
+        cxx_base_specifier_list = [
+            sub_item["name"]
+            for sub_item in self.members
+            if sub_item["kind"] == "CXX_BASE_SPECIFIER"
+        ]
+        if cxx_base_specifier_list:
+            cxx_base_specifier_list = ",".join(cxx_base_specifier_list)
+            self.linelist.append(
+                f'py::class_<{self.name}, {cxx_base_specifier_list.replace("struct ", "").replace("pcl::", "")}>(m, "{self.name}")'
+            )
+        else:
+            self.linelist.append(f'py::class_<{self.name}>(m, "{self.name}")')
 
-def handle_operator(item):
-    pass
-    # if in_struct:
-    #     module_linelist.append(f'.def(py::self {item["identifier"]} py::self)')
+        # TODO: Merge this and next block via a design updation
+        # handle anonymous structs, etc. as field declarations
+        for sub_item in self.members:
+            fields = self.get_fields_from_anonymous(sub_item)
+            for field in fields:
+                if field["element_type"] == "ConstantArray":
+                    # TODO: FIX: readwrite, not readonly
+                    self.linelist.append(
+                        f'.def_property_readonly("{field["name"]}", []({self.name}& obj) {{return obj.{field["name"]}; }})'  # float[ ' + f'obj.{sub_item["name"]}' + '.size()];} )'
+                    )
+                else:
+                    self.linelist.append(
+                        f'.def_readwrite("{field["name"]}", &{self.name}::{field["name"]})'
```

Comment:
I get a feeling that this could be 2 separate functions:
* handle_anon
* handle_fields

and you'd just call them like:
```python
for elem in self.members:
    self.kind_functions[elem["kind"](elem)
```